src/Event.php

```diff
@@ -154,6 +166,58 @@ public function getId(): string
         return str_replace('-', '', $this->id->toString());
     }
 
+    /**
+     * Gets the identifier of the SDK package that generated this event.
+     *
+     * @return string
+     *
+     * @internal
+     */
+    public function getSdkIdentifier(): string
+    {
+        return $this->sdkIdentifier;
+    }
+
+    /**
+     * Sets the identifier of the SDK package that generated this event.
+     *
+     * @param string $sdkIdentifier
+     *
+     * @internal
+     */
+    public function setSdkIdentifier(string $sdkIdentifier): void
+    {
+        $this->sdkIdentifier = $sdkIdentifier;
+    }
+
+    /**
+     * Gets the version of the SDK package that generated this Event.
+     *
+     * @return string
+     *
+     * @internal
+     */
+    public function getSdkVersion(): string
+    {
+        if (null === $this->sdkVersion) {
+            $this->sdkVersion = PrettyVersions::getVersion('sentry/sentry')->getPrettyVersion();
```

Comment:
I don't understand why this exact code is duplicated in the `ClientBuilder` class. I thought that the purpose of the event factory was also to set this attribute, we should not have this logic inside the `Event` class (the attribute is not even required so it can be left `nullable` without any sort of problem)

Reply:
I left it here to provide a sensible default. If this is not required, I can leave it as nullable and remove this piece. @HazAT WDYT?